external/fv3fit/fv3fit/_shared/io.py

```diff
@@ -18,17 +18,20 @@ class _Register:
 
     def __init__(self) -> None:
         self._model_types: MutableMapping[str, Type[Predictor]] = {}
+        self._dump_types: MutableMapping[str, Type[Dumpable]] = {}
 
-    def __call__(self, name: str) -> Callable[[Type[Predictor]], Type[Predictor]]:
-        if name in self._model_types:
+    def __call__(self, name: str) -> Callable[[Type[Dumpable]], Type[Dumpable]]:
```

Comment:
Is this added complexity needed? Given the current definitions of `Predictor`, `Loadable`, and `Dumpable`, it seems like the intuitive thing to be registering is `Predictor`. What's the value of registering something that can only be dumped but not loaded?

Reply:
Reverted changes to this file and instead set it to dump/load a new Reloadable ABC inherited by Predictor.

Reply:
Yeah, as discussed a `Predictor` is something that can be dumped/loaded and that can `.predict()`. Beyond that I personally didn't expect specific behavior of the predict method.